src/libUtils/Logger.cpp

```diff
@@ -16,325 +16,262 @@
  */
 
 #include "Logger.h"
+#include "libUtils/TimeUtils.h"
 
-#include <pthread.h>
-#include <sys/syscall.h>
-#include <unistd.h>
-#include <boost/filesystem.hpp>
-#include <boost/format.hpp>
-#include <cstring>
-#include <iostream>
+#include <g3sinks/LogRotate.h>
+
+#include <boost/filesystem/operations.hpp>
 
 using namespace std;
 using namespace g3;
 
-inline string MyCustomFormatting(const LogMessage& msg) {
-  return string("[") + msg.level().substr(0, 4) + "]";
+namespace {
+
+#define LIMIT(s, len)                              \
+  std::setw(len) << std::setfill(' ') << std::left \
+                 << std::string(s).substr(0, len)
+
+#define LIMIT_RIGHT(s, len)                        \
+  std::setw(len) << std::setfill(' ') << std::left \
+                 << s.substr(std::max<int>((int)s.size() - len, 0))
+
+// Auxiliary class to write manipulators for log message formatting.
+template <std::ostream& (*FuncT)(std::ostream&, const LogMessage&)>
+struct LoggerManip {
+  LoggerManip(const LogMessage& message) : m_message{message} {}
+
+  std::ostream& operator()(std::ostream& stream) const {
+    return FuncT(stream, m_message);
+  }
+
+ private:
+  const LogMessage& m_message;
+};
+}  // namespace
+
+namespace std {
+// Output operator to be able to use LoggerManip manipulators.
+template <std::ostream& (*FuncT)(std::ostream&, const LogMessage&)>
+inline ostream& operator<<(std::ostream& stream,
+                           const LoggerManip<FuncT>& manip) {
+  return manip(stream);
 }
+}  // namespace std
 
 namespace {
-/// helper function to get tid with better cross-platform support
-inline pid_t getCurrentPid() {
-#if defined(__linux__)
-  return syscall(SYS_gettid);
-#elif defined(__APPLE__) && defined(__MACH__)
-  uint64_t tid64;
-  pthread_threadid_np(NULL, &tid64);
-  return static_cast<pid_t>(tid64);
-#else
-#error  // not implemented in this platform
-  return 0;
-#endif
+std::ostream& logThreadId(std::ostream& stream, const LogMessage& message) {
+  stream << '[' << std::hex
+         << PAD(message._call_thread_id, Logger::TID_LEN, ' ') << std::dec
+         << ']';
+  return stream;
 }
-}  // namespace
 
-const streampos Logger::MAX_FILE_SIZE =
-    1024 * 1024 * 100;  // 100MB per log file
+std::ostream& logTimestamp(std::ostream& stream, const LogMessage& message) {
+  // The following is taken from g3log's localtime_formatted; we're changing
+  // it here to show our own time format in UTC instead of localtime.
```

Comment:
Add a link and mention the license? We can't accidentally re-license it under GPL :)

Reply:
The license is completely free. Can be checked in the repo but I've added the link.